lib/pf/profile/filter/day_of_week.pm

```diff
@@ -0,0 +1,101 @@
+package pf::profile::filter::day_of_week;
+=head1 NAME
+
+pf::profile::filter::day_of_week add documentation
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::profile::filter::day_of_week
+
+=cut
+
+use strict;
+use warnings;
+use Date::Format;
+use Moo;
+extends 'pf::profile::filter';
+
+=head1 ATTRIBUTES
+
+=head2 allowed_days
+
+The allowed days of the week
+
+=cut
+
+has allowed_days => ( is => 'rw' );
+
+=head2 value
+
+add a trigger to the value to create update allowed_days
+
+=cut
+
+has '+value' => ( trigger => 1 );
+
+=head1 METHODS
+
+=head2 match
+
+    Return true if the current day of week matches the values provided
+    The value is expected to be in the following format
+    1,3,5
+    Where the number is the day of the week
+    1 - Monday
+    2 - Tuesday
+    3 - Wednesday
+    4 - Thursday
+    5 - Friday
+    6 - Saturday
+    7 - Sunday
+
+=cut
+
+sub match {
+    my ($self) = @_;
+    my $current = time2str("%W",time);
+    return exists ${$self->allowed_days}{$current};
+}
+
+=head2 _trigger_value
+
+Set allowed_days from the value
+
+=cut
+
+sub _trigger_value {
+    my ($self) = @_;
+    my %allowed_days = map { $_ => undef} split /\s*,\s*/ ,$self->value;
```

Comment:
Could we maybe create that hash with values of 1. Just to make it explicit that it's a truth value.